pkg/analysis/helpers/markers/analyzer_test.go

```diff
@@ -150,103 +152,302 @@ func TestExtractMarkerIdAndExpressions(t *testing.T) {
+				}
+				return nil
+			},
 		},
 		{
-			name:           "valid marker - with backtickets and parentheses",
-			comment:        "// +k8s:ifEnabled(`foo`)=+k8s:required",
-			shouldBeMarker: true,
-			expectedID:     "k8s:ifEnabled(`foo`)",
+			name:    "invalid marker - starts with special characters",
+			comment: &ast.Comment{Text: "// +!*@(#&KSDJUF:A"},
+			expect: func(m Marker) error {
+				expectedID := ""
+				if m.Identifier != expectedID {
+					return fmt.Errorf("marker identifier %q does not match expected %q", m.Identifier, expectedID)
+				}
+				return nil
+			},
 		},
 		{
-			name:           "invalid marker - markdown table border",
-			comment:        "// +-------+-------+-------+",
-			shouldBeMarker: false,
-			expectedID:     "",
+			name:    "regular comment - no plus sign",
+			comment: &ast.Comment{Text: "// This is a regular comment"},
+			expect: func(m Marker) error {
+				expectedID := ""
+				if m.Identifier != expectedID {
+					return fmt.Errorf("marker identifier %q does not match expected %q", m.Identifier, expectedID)
+				}
+				return nil
+			},
 		},
+
 		{
-			name:           "invalid marker - markdown table border without pipes",
-			comment:        "// +----------",
-			shouldBeMarker: false,
-			expectedID:     "",
+			name: "simple declarative validation marker",
+			comment: &ast.Comment{
+				Text: "// +k8s:required",
+			},
+			expect: func(m Marker) error {
+				if m.Identifier != "k8s:required" {
+					return fmt.Errorf("identifier %q did not match expected identifier %q", m.Identifier, "k8s:required")
+				}
+
+				return nil
+			},
 		},
 		{
-			name:           "invalid marker - starts with special characters",
-			comment:        "// +!*@(#&KSDJUF:A",
-			shouldBeMarker: false,
-			expectedID:     "",
+			name: "declarative validation marker with a value",
+			comment: &ast.Comment{
+				Text: "// +k8s:maxLength=10",
+			},
+			expect: func(m Marker) error {
+				if m.Identifier != "k8s:maxLength" {
+					return fmt.Errorf("identifier %q did not match expected identifier %q", m.Identifier, "k8s:required")
+				}
+
+				payload, ok := m.Expressions["payload"]
+				if !ok {
+					return errors.New("expected a payload but there was none")
+				}
+
+				if payload != "10" {
+					return fmt.Errorf("payload value of %q does not match expected value %q", payload, "10")
+				}
+
+				return nil
+			},
 		},
 		{
-			name:           "regular comment - no plus sign",
-			comment:        "// This is a regular comment",
-			shouldBeMarker: false,
-			expectedID:     "",
+			name: "declarative validation marker with named argument",
+			comment: &ast.Comment{
+				Text: "// +k8s:unionMember(union: \"union1\")",
+			},
+			expect: func(m Marker) error {
+				if m.Identifier != "k8s:unionMember" {
+					return fmt.Errorf("identifier %q did not match expected identifier %q", m.Identifier, "k8s:unionMember")
+				}
+
+				union, ok := m.Expressions["union"]
```

Comment:
I can't help but feel we are now mixing two different kinds of args (those in payload vs those in the args/parentheses) and that these should be expressed differently, else we can't reliably reconstruct right?